libraries/AP_Common/AP_FWVersion.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <stdint.h>
+
+typedef struct {
+    uint8_t major;
+    uint8_t minor;
+    uint8_t patch;
+    uint8_t fw_type; // this comes from mavlink enumeration
```

Comment:
I was trying to avoid pulling in the mavlink header.

Changed to use `FIRWMARE_VERSION_TYPE`